### PR TITLE
Document updating requirements with pre commit hooks

### LINKED ISSUE #1335: Document how to check multiple requirements with pre-commit hooks
> Thanks for the issue! You should use a separate hook for each input requirement file, for example:
> 
> ```yaml
> repos:
> -   repo: https://github.com/jazzband/pip-tools
>     rev: 5.3.1
>     hooks:
>       - id: pip-compile
>         name: pip-compile setup.py
>         files: ^setup\.py$
>       - id: pip-compile
>         name: pip-compile requirements-dev.in
>         args: [requirements-dev.in]
>         files: ^requirements-dev\.(in|txt)$
>       - id: pip-compile
>         name: pip-compile requirements-lint.in
>         args: [requirements-lint.in]
>         files: ^requirements-lint\.(in|txt)$
> ```

Would it be possible to add this somewhere in the documentation? I wasn't familiar at all with how pre-commit really worked so it took some digging until I realized that even if the pattern matches for `files` we would still need to feed the names in individually for each requirements file for it to work as intended.

_Originally posted by @seanytak in https://github.com/jazzband/pip-tools/issues/1208#issuecomment-784675891_

### PR DESCRIPTION
**Pull request by microcat49**
Resolves #1335 

**Changelog-friendly one-liner**: 

This commit adds documentation about using pre-commit when a user has multiple requiments. A user need to have multiple hooks for each requirements file and this commit makes that informationn explicit in the documentation.

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -358,6 +358,26 @@ You might want to customize ``pip-compile`` args by configuring ``args`` and/or
...
+
+.. code-block:: yaml 
+
+    repos:
+    -   repo: https://github.com/jazzband/pip-tools
+        rev: 5.3.1
+        hooks:
+          - id: pip-compile
+            name: pip-compile setup.py
+            files: ^setup\.py$
```
Comment (atugushev) [inline README.rst:371]: I think we should include `requirements.txt` also here.

**Comment by microcat49** [comment]
Hey @atugushev, sorry about the delayed response on this one. This is ready for another review.

File: README.rst (hunk 1 above)
Comment (atugushev) [reply README.rst:371]: I mean something like:

```yaml
files: ^(setup\.py|requirements\.txt)$
```

**Review by ssbarnea — APPROVED** [review]

File: README.rst [hunk 2]
```diff
@@ -368,6 +368,30 @@ You might want to customize ``pip-compile`` args by configuring ``args`` and/or
             files: ^requirements/production\.(in|txt)$
             args: [--index-url=https://example.com, requirements/production.in]
 
+If you have multiple requirement files make sure you create a hook for each file. 
+
+.. code-block:: yaml 
+
+    repos:
+    -   repo: https://github.com/jazzband/pip-tools
```
Comment (webknjaz) [inline README.rst:376]: This is inconsistent with the style of the snippets above.

**Review by ssbarnea — APPROVED** [review]

**Pull request merged**